src/SessionMonitor/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { userapiPath } from '../localconf';
+import getReduxStore from '../reduxStore';
+import { logoutAPI } from '../actions';
+
+class SessionMonitor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { mostRecentActivityTimestamp: Date.now() };
+  }
+
+  componentDidMount() {
+    window.addEventListener('mousedown', this.updateUserActivity, false);
```

Comment:
If we're making a component out of this, then we should have logic to remove the window listeners and cancel the setInterval when the component unmounts, and also a guard against having multiple copies of this component on the page.  This thing is really a utility - not a UX component: `sessionMonitor.start()`, `sessionMonitor.end()` - but I get that people love to make a component out of everything :-p